alephclient/api.py

```diff
@@ -353,6 +353,32 @@ def _bulk_chunk(
                     return
                 backoff(ae, attempt)
 
+    def write_entity(self, collection_id: str, entity: Iterable, **kw):
+        """Create a single entity via the API, in the given
+        collection.
+
+        params
+        ------
+        collection_id: id of the collection to use. This will overwrite any
+        existing collection specified in the entity
+        entity: A dict object containing the values of the entity
+        """
+        if isinstance(entity, Dict):
+            entity["collection_id"] = collection_id
+
+        for attempt in count(1):
+            url = self._make_url("entities")
+            try:
+                response = self.session.post(url, json=entity)
+                response.raise_for_status()
+                return
+            except RequestException as exc:
+                ae = AlephException(exc)
+                if not ae.transient or attempt > self.retries:
+                    log.error(ae)
```

Comment:
Hmmm this almost feels like it wants to raise, no?

Reply:
I agree. Otherwise the worker may not know a step has failed.

Reply:
Didn't write this so flying blind but I'll get this to raise a request exception after it has logged the error.

Reply:
I think I might have refactored this... my concern at the time was that throwing an exception would simply kill the worker thread and as the command processes a directory, all the workers will die off. So you'd either have to have some logic to re-spawn workers, or catch the exception further up but still inside the worker.

Reply:
Hi @mattcg, I think you're talking about the code in `crawldir` where there will me many calls to upload all the files. And I agree there should be an option to be more error tolerant in that case.

Here we're uploading a single entity only. So IMO it's safe to just raise the error and fail immediately.

Reply:
Ah you're right. Sorry, got confused.